markdown/src/main/java/it/niedermann/android/markdown/MarkdownUtil.java

```diff
@@ -438,24 +438,18 @@ public static String getMarkdownLink(@NonNull String text, @NonNull String url)
      * @return Plain text string
      */
     @NonNull
+    // CS304 issue link: https://github.com/stefan-niedermann/nextcloud-notes/issues/1212
     public static String removeMarkdown(@Nullable String s) {
         if (s == null)
             return "";
         // TODO maybe we can utilize the markwon renderer?
-
-        for (EListType listType : EListType.values()) {
-            for (String item : Arrays.asList(listType.checkboxChecked, listType.checkboxUnchecked, listType.listSymbolWithTrailingSpace)) {
-                if (s.startsWith(item)) {
-                    s = s.substring(item.length());
-                }
-            }
-        }
-        s = PATTERN_LISTS.matcher(s).replaceAll("");
-        s = PATTERN_HEADINGS.matcher(s).replaceAll("$1");
-        s = PATTERN_HEADING_LINE.matcher(s).replaceAll("");
-        s = PATTERN_EMPHASIS.matcher(s).replaceAll("$2");
-        s = PATTERN_SPACE_1.matcher(s).replaceAll("");
-        s = PATTERN_SPACE_2.matcher(s).replaceAll("");
-        return s;
+        // Create HTML string from Markup
+        String html = renderer.render(parser.parse(replaceCheckboxesWithEmojis(s)));
+        html = html.replaceAll("\n","");
+        // Create Spanned from HTML, with special handling for ordered list items
+        Spanned spanned = Html.fromHtml(html,Html.FROM_HTML_MODE_COMPACT);
```

Comment:
Funny thing that you mentioned a "special handling for ordered list items" but don't apply it :smile: I guess copy & paste?
Just pass in `null` as third argument and `new ListTagHandler` as fourth argument to `fromHtml`.